airbyte-cdk/python/airbyte_cdk/sources/declarative/async_job/job_orchestrator.py

```diff
@@ -226,7 +224,12 @@ def create_and_get_completed_partitions(self) -> Iterable[AsyncPartition]:
             Each partition is wrapped in an Optional, allowing for None values.
         """
         while True:
-            self._start_jobs()
+            try:
+                self._start_jobs()
+            except self._exceptions_to_break_on as e:
+                raise e
+            except AirbyteTracedException as e:
```

Comment:
Should we catch any exception here and not just AirbyteTracedException?

Reply:
all exceptions from httpclient are wrapped in `AirbyteTracedException`, 
for all the rest I'd like them to be raised for now, unit I don't know where they may come from

Reply:
The issue is that the JobOrchestrator does not know which implementation of the JobRepository that is used. Right now, maybe all of these are AirbyteTracedExceptions but if we add a FacebookMarketingJobRepository or a Grpc one, the errors raised might be different and it'll require us to either know we have to update the JobOrchestrator or know we have to only emit AirbyteTracedException.

Is there a case you had in mind that you were worried about?

Reply:
refactored.

> Is there a case you had in mind that you were worried about?

Nothing particular on my mind. Just don't like `except Exception:` line